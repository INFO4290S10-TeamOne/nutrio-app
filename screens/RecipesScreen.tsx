import { View, ScrollView } from 'react-native';
import SearchBar from '../components/SearchBar';
import CuisineShortcut from '../components/CuisineShortcut';
import RecipeCard from '../components/RecipeCard';

const cuisines = [
  { id: 1, title: 'Italian', image: require('../assets/favicon.png') },
  { id: 2, title: 'Chinese', image: require('../assets/favicon.png') },
  { id: 3, title: 'Sushi', image: require('../assets/favicon.png') },
  { id: 4, title: 'American', image: require('../assets/favicon.png') },
  { id: 5, title: 'Chicken', image: require('../assets/favicon.png') },
  { id: 6, title: 'Burger', image: require('../assets/favicon.png') },
  { id: 7, title: 'Desserts', image: require('../assets/favicon.png') },
  { id: 8, title: 'Asian', image: require('../assets/favicon.png') },
];

const recipes = [
  {
    id: 1,
    title: 'Chicken Parmesan',
    image: require('../assets/favicon.png'),
    time: 30,
    servings: 4,
  },
  {
    id: 2,
    title: 'Chicken Parmesan',
    image: require('../assets/favicon.png'),
    time: 30,
    servings: 4,
  },
  {
    id: 3,
    title: 'Chicken Parmesan',
    image: require('../assets/favicon.png'),
    time: 30,
    servings: 4,
  },
];

const RecipesScreen = () => {
  return (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <SearchBar />
      <ScrollView>
        <ScrollView horizontal={true} showsHorizontalScrollIndicator={false}>
          {cuisines.map((cuisine) => (
            <CuisineShortcut key={cuisine.id} title={cuisine.title} image={cuisine.image} />
          ))}
        </ScrollView>
        {/* Divider */}
        <View
          style={{
            borderBottomColor: '#ccc',
            borderBottomWidth: 1,
            width: '90%',
            alignSelf: 'center',
            marginVertical: 20,
          }}
        />

        {/* // Recipe cards */}
        {recipes.map((recipe) => (
          <RecipeCard
            key={recipe.id}
            title={recipe.title}
            image={recipe.image}
            time={recipe.time}
            servings={recipe.servings}
          />
        ))}
      </ScrollView>
    </View>
  );
};

export default RecipesScreen;
