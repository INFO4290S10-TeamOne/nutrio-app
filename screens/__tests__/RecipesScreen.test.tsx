import React from 'react';
import renderer from 'react-test-renderer';

import RecipesScreen from '../RecipesScreen';

it('renders the recipes screen', () => {
  const tree = renderer.create(<RecipesScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
