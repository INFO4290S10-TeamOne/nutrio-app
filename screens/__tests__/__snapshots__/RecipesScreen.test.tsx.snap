// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the recipes screen 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#F0EEEE",
        "borderRadius": 15,
        "flexDirection": "row",
        "height": 50,
        "marginBottom": 10,
        "marginTop": 15,
        "paddingHorizontal": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "width": "90%",
      }
    }
  >
    <TextInput
      placeholder="Search..."
      style={
        {
          "flex": 1,
          "fontSize": 18,
        }
      }
      testID="searchBar"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="searchButton"
    >
      <
        color="gray"
        icon={
          {
            "icon": [
              512,
              512,
              [
                128269,
                "search",
              ],
              "f002",
              "M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z",
            ],
            "iconName": "magnifying-glass",
            "prefix": "fas",
          }
        }
        size={24}
      />
    </View>
  </View>
  <RCTScrollView>
    <View>
      <RCTScrollView
        horizontal={true}
        showsHorizontalScrollIndicator={false}
      >
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 50,
                "elevation": 5,
                "height": 80,
                "justifyContent": "center",
                "margin": 10,
                "opacity": 1,
                "position": "relative",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 80,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 50,
                  "resizeMode": "cover",
                  "width": 50,
                }
              }
            />
            <Text
              style={
                {
                  "bottom": 5,
                  "fontSize": 14,
                  "position": "absolute",
                }
              }
            >
              Italian
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 50,
                "elevation": 5,
                "height": 80,
                "justifyContent": "center",
                "margin": 10,
                "opacity": 1,
                "position": "relative",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 80,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 50,
                  "resizeMode": "cover",
                  "width": 50,
                }
              }
            />
            <Text
              style={
                {
                  "bottom": 5,
                  "fontSize": 14,
                  "position": "absolute",
                }
              }
            >
              Chinese
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 50,
                "elevation": 5,
                "height": 80,
                "justifyContent": "center",
                "margin": 10,
                "opacity": 1,
                "position": "relative",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 80,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 50,
                  "resizeMode": "cover",
                  "width": 50,
                }
              }
            />
            <Text
              style={
                {
                  "bottom": 5,
                  "fontSize": 14,
                  "position": "absolute",
                }
              }
            >
              Sushi
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 50,
                "elevation": 5,
                "height": 80,
                "justifyContent": "center",
                "margin": 10,
                "opacity": 1,
                "position": "relative",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 80,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 50,
                  "resizeMode": "cover",
                  "width": 50,
                }
              }
            />
            <Text
              style={
                {
                  "bottom": 5,
                  "fontSize": 14,
                  "position": "absolute",
                }
              }
            >
              American
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 50,
                "elevation": 5,
                "height": 80,
                "justifyContent": "center",
                "margin": 10,
                "opacity": 1,
                "position": "relative",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 80,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 50,
                  "resizeMode": "cover",
                  "width": 50,
                }
              }
            />
            <Text
              style={
                {
                  "bottom": 5,
                  "fontSize": 14,
                  "position": "absolute",
                }
              }
            >
              Chicken
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 50,
                "elevation": 5,
                "height": 80,
                "justifyContent": "center",
                "margin": 10,
                "opacity": 1,
                "position": "relative",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 80,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 50,
                  "resizeMode": "cover",
                  "width": 50,
                }
              }
            />
            <Text
              style={
                {
                  "bottom": 5,
                  "fontSize": 14,
                  "position": "absolute",
                }
              }
            >
              Burger
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 50,
                "elevation": 5,
                "height": 80,
                "justifyContent": "center",
                "margin": 10,
                "opacity": 1,
                "position": "relative",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 80,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 50,
                  "resizeMode": "cover",
                  "width": 50,
                }
              }
            />
            <Text
              style={
                {
                  "bottom": 5,
                  "fontSize": 14,
                  "position": "absolute",
                }
              }
            >
              Desserts
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 50,
                "elevation": 5,
                "height": 80,
                "justifyContent": "center",
                "margin": 10,
                "opacity": 1,
                "position": "relative",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 80,
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 50,
                  "resizeMode": "cover",
                  "width": 50,
                }
              }
            />
            <Text
              style={
                {
                  "bottom": 5,
                  "fontSize": 14,
                  "position": "absolute",
                }
              }
            >
              Asian
            </Text>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          {
            "alignSelf": "center",
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "marginVertical": 20,
            "width": "90%",
          }
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "center",
            "backgroundColor": "#fff",
            "borderRadius": 15,
            "elevation": 5,
            "height": 220,
            "marginBottom": 20,
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": "90%",
          }
        }
        testID="recipe-card"
      >
        <Image
          source={1}
          style={
            {
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "height": 150,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
              }
            }
          >
            Chicken Parmesan
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                {
                  "color": "gray",
                  "fontSize": 14,
                }
              }
            >
              Time: 
              30
               min
            </Text>
            <Text
              style={
                {
                  "color": "gray",
                  "fontSize": 14,
                }
              }
            >
              Servings: 
              4
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "center",
            "backgroundColor": "#fff",
            "borderRadius": 15,
            "elevation": 5,
            "height": 220,
            "marginBottom": 20,
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": "90%",
          }
        }
        testID="recipe-card"
      >
        <Image
          source={1}
          style={
            {
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "height": 150,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
              }
            }
          >
            Chicken Parmesan
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                {
                  "color": "gray",
                  "fontSize": 14,
                }
              }
            >
              Time: 
              30
               min
            </Text>
            <Text
              style={
                {
                  "color": "gray",
                  "fontSize": 14,
                }
              }
            >
              Servings: 
              4
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "center",
            "backgroundColor": "#fff",
            "borderRadius": 15,
            "elevation": 5,
            "height": 220,
            "marginBottom": 20,
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": "90%",
          }
        }
        testID="recipe-card"
      >
        <Image
          source={1}
          style={
            {
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "height": 150,
              "width": "100%",
            }
          }
        />
        <View
          style={
            {
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 10,
              }
            }
          >
            Chicken Parmesan
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                {
                  "color": "gray",
                  "fontSize": 14,
                }
              }
            >
              Time: 
              30
               min
            </Text>
            <Text
              style={
                {
                  "color": "gray",
                  "fontSize": 14,
                }
              }
            >
              Servings: 
              4
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
